internal/document_test.go

```diff
@@ -74,3 +113,448 @@ func TestDocument_Debug(t *testing.T) {
+		})
+
+		t.Run("unknown signing authority - nil roots", func(t *testing.T) {
+			t.Log(key)
+
+			// given
+			certProvider := mocks.NewInternalCertProvider(t)
+
+			// expecting
+			certProvider.EXPECT().Roots().Return(nil, nil)
+
+			// when
+			_, _, err := documents[key].doc.CheckCertificates(
+				certProvider,
+				internal.WithAttestationTime(),
+			)
+
+			// then
+			assert.ErrorContains(t, err, "verifying certificate")
+		})
+
+		t.Run("unknown signing authority - empty roots", func(t *testing.T) {
+			t.Log(key)
+
+			// given
+			certProvider := mocks.NewInternalCertProvider(t)
+
+			// expecting
+			certProvider.EXPECT().Roots().Return(x509.NewCertPool(), nil)
+
+			// when
+			_, _, err := documents[key].doc.CheckCertificates(
+				certProvider,
+				internal.WithAttestationTime(),
+			)
+
+			// then
+			assert.ErrorContains(t, err, "verifying certificate")
+		})
+
+		timeOutOfBoundsTests := []struct {
+			name        string
+			timeOpt     internal.VerificationTimeFunc
+			errContains string
+		}{
+			{
+				"certificate expired",
+				internal.WithTime(time.Date(10000, 0, 0, 0, 0, 0, 0, time.UTC)),
+				"verifying certificate",
+			},
+			{
+				"certificate not yet valid",
+				internal.WithTime(time.Date(1970, 0, 0, 0, 0, 0, 0, time.UTC)),
+				"verifying certificate",
+			},
+			{
+				"zero time",
+				internal.WithTime(time.Time{}),
+				"verification time is 0",
+			},
+		}
+		for _, tt := range timeOutOfBoundsTests {
+			t.Run(tt.name, func(t *testing.T) {
+				t.Log(key)
+
+				// when
+				_, _, err := documents[key].doc.CheckCertificates(
+					documents[key].certProvider,
+					tt.timeOpt,
+				)
+
+				// then
+				assert.ErrorContains(t, err, tt.errContains)
+			})
+		}
+	}
+
+	t.Run("parsing cert", func(t *testing.T) {
+		// given
+		doc := internal.Document{}
+
+		// when
+		_, _, err := doc.CheckCertificates(nil, nil)
+
+		// then
+		assert.ErrorContains(t, err, "parsing cert")
+	})
+
+	t.Run("missing cabundle", func(t *testing.T) {
+		// given
+		doc := nitroDoc
+		doc.CABundle = nil
+
+		// when
+		_, _, err := doc.CheckCertificates(nil, nil)
+
+		// then
+		assert.ErrorContains(t, err, "missing cabundle")
+	})
+
+	t.Run("missing cabundle item", func(t *testing.T) {
+		// given
+		doc := nitroDoc
+		doc.CABundle = [][]byte{nil}
+
+		// when
+		_, _, err := doc.CheckCertificates(nil, nil)
+
+		// then
+		assert.ErrorContains(t, err, "cabundle item")
+	})
+
+	t.Run("cabundle item too short", func(t *testing.T) {
+		// given
+		doc := nitroDoc
+		doc.CABundle = [][]byte{make([]byte, 0)}
+
+		// when
+		_, _, err := doc.CheckCertificates(nil, nil)
+
+		// then
+		assert.ErrorContains(t, err, "cabundle item")
+	})
+
+	t.Run("cabundle item too long", func(t *testing.T) {
+		// given
+		doc := nitroDoc
+		doc.CABundle = [][]byte{make([]byte, 1025)}
+
+		// when
+		_, _, err := doc.CheckCertificates(nil, nil)
+
+		// then
+		assert.ErrorContains(t, err, "cabundle item")
```

Comment:
Minor, but `cabundle item` is not all that informative, so that the test is not that clear on its own. Maybe use more of the error string, since you know how it is going to render?

Reply:
Updated to include more of the expected error string so you can tell which error case is getting triggered